### PR TITLE
A new documentation page and layout fixes for KBreadcrumbs

### PR DESCRIPTION
**Pull request by MisRob**
### (2) Consolidates styles in regards to vertical alignment of breadcrumbs items

|  | Before | After |
| - | -------- | ------- |
| KDS docs | doc-breadcrumbs-vertical-align-before | doc-breadcrumbs-vertical-align-after |

### (3) Optimizes breadcrumbs to use all space available

|  | Before | After |
| - | -------- | ------- |
| KDS docs | doc-breadcrumbs-space-before | doc-breadcrumbs-space-after |
| Kolibri | kolibri-breadcrumbs-before | kolibri-breadcrumbs-after |

### (4) Consolidates throttling and resizing logic

Particularly in regards to having more breadcrumbs instances on one page like in KDS documentation. See commit messages and code comments for details. In the following recording, notice the second breadcrumbs not being properly collapsed because `updateCrumbs` never ran for this particular instance:

|  | Before | After |
| - | -------- | ------- |
| KDS docs | doc-breadcrumbs-throttle-before | doc-breadcrumbs-throttle-after |

#### Issue addressed

- Resolves [https://github.com/learningequality/kolibri-design-system/issues/284](https://github.com/learningequality/kolibri-design-system/issues/284)
- Resolves [https://github.com/learningequality/kolibri-design-system/issues/198](https://github.com/learningequality/kolibri-design-system/issues/198)
- Resolves [https://github.com/learningequality/kolibri/issues/6918](https://github.com/learningequality/kolibri/issues/6918)
- Resolves https://github.com/learningequality/kolibri-design-system/issues/129

### The documentation page

Regarding text, this is just the very first draft that’s main purpose was to provide me with testing and development environment for new updates as @indirectlylit suggested:

> I realized that a simple alternative to Story Book integration is to simply expand the number of examples we embed in the `K*` pages. In this case for example I was able to update and test the changes without ever leaving the docs

Feel free to suggest any changes. @jtamiace I’d appreciate particularly your thoughts. Also, @jtamiace, I’d like to add some guidance to the new page on when to allow displaying a single breadcrumb item ([https://github.com/learningequality/kolibri-design-system/issues/129](https://github.com/learningequality/kolibri-design-system/issues/129)) on this opportunity. Do you have any suggestions?

### The library component

- [ ]  Check breadcrumbs while resizing window, collapsing of items and items alignment
- [ ]  Test breadcrumbs on [the new KDS documentation page](https://deploy-preview-292--kolibri-design-system.netlify.app/kbreadcrumbs/)
- [ ]  Test breadcrumbs in the corresponding Kolibri PR https://github.com/learningequality/kolibri/pull/8972
  - Confirm that [https://github.com/learningequality/kolibri/issues/6918](https://github.com/learningequality/kolibri/issues/6918) has been fixed
  - Check for breadcrumbs regressions in the whole app

## Testing checklist

- [x] Contributor has fully tested the PR manually
- [x] If there are any front-end changes, before/after screenshots are included
- [ ] The change has been added to the `changelog`

## Reviewer guidance

- [ ] Is the code clean and well-commented?
- [ ] Are all UI components LTR and RTL compliant (if applicable)?

## Post-merge updates and tracking

- [ ] Learning Platform update PR is submitted
- [ ] Learning Platform update PR is merged
- [ ] Studio update PR is submitted
- [ ] Studio update PR is merged
- [ ] Data Portal update PR is submitted
- [ ] Data Portal update PR is merged

## Comments

I’ll update the changelog and create a new release tag as soon as the first review of the documentation page and the component is done.

### CODE REVIEW

File: lib/KBreadcrumbs.vue [hunk 1]
```diff
@@ -182,79 +185,89 @@
...
-      this.attachSensors();
-      this.$watch('parentWidth', this.throttleUpdateCrumbs);
-    },
-
-    beforeDestroy() {
-      this.detachSensors();
+      this.attachRefs();
+      // The throttled update function is defined here and not as a method on purpose
+      // since having it defined as a method on the options object would cause problems
+      // when there are more KBreadcrumbs instances rendered on one page.
```
Comment (nucleogenesis) [inline lib/KBreadcrumbs.vue:198]: This makes sense thanks for the explanation!

**Review by nucleogenesis — APPROVED** [review]
Code looks good. I love the documentation. Since this is documented directly on the component page, I wonder if we might consider a design that shows code example along with the example. Looking at the code for the docs component I think it'd be nice to see directly in the docs. (I'll tuck this thought away for an upcoming tactical so no need to do anything here). I didn't spin it up because it looks good in your screenshots and on the live netlify of the docs.

**Comment by MisRob** [comment]
@nucleogenesis

>  wonder if we might consider a design that shows code example along with the example

Yes, I was thinking about exactly the same thing. I welcome chatting about it during the tactical as you suggest since it's related rather to the whole KDS documentation.

**Comment by jtamiace** [comment]
> I’d like to add some guidance to the new page on when to allow displaying a single breadcrumb item (#129) on this opportunity.

The only situation I can think of that would be useful to show the single breadcrumb is when there isn't any other page header or label that shows what the current level is.

**Comment by indirectlylit** [comment]
> wonder if we might consider a design that shows code example along with the example

Hi friends! Thought I'd provide a little context here:

The [`DocsShowCode`](https://github.com/learningequality/kolibri-design-system/blob/main/docs/common/DocsShowCode.vue) component can be useful. The new [layout page](https://design-system.learningequality.org/layout/)  has some instances of showing code + examples.

However, the component does not actually render the example code – that is a separate block.

This is somewhat unfortunate because the code is duplicated. On the other hand, it can be useful because some examples need additional code (e.g. css margins) that is irrelevant to demonstrating the component API and this can be omitted.

Another option and something that could be useful regardless: changing the Github link into a "View source" link for the current page, which would let devs more easily inspect the source directly kind of like on readthedocs:

| kds | rtd |
|--|--|
| image | image |


Also some history:

At one point we actually had a system working where the code displayed would drive a live example, and you could even edit the code to see changes. This ended up being challenging because it required the Vue template compiler to be loaded client-side and it required the entire template to shown including the entire `<template />` and `<script />` tags which was distracting. If these challenges can be overcome that could be neat, but I personally gave up on it in favor of the simpler `DocsShowCode`.

**Comment by MisRob** [comment]
Thank you, @indirectlylit.

@jtamiace I added your suggestion on allowing a single breadcrumb to the guidelines section. Could you please review the whole page?

**Comment by MisRob** [comment]
@jtamiace The latest preview is here https://deploy-preview-292--kolibri-design-system.netlify.app/kbreadcrumbs/

**Review by jtamiace — APPROVED** [review]
Thank you, this new documentation is great

**Review by marcellamaki — APPROVED** [review]

**Pull request merged**